### PR TITLE
LOK-2401: DevOps: Secure the Prometheus expression browser

### PR DESCRIPTION
**Pull request by mershad-manesh**

### CODE REVIEW

**Review by MuhammadTalhaDataQHealth — COMMENTED** [review]
values are not mentioned correctly. This should be changed to

`{{- if not .Values.prometheus.auth.externalSecret  }}
{{- $user := required "Prometheus.auth.username required" .Values.prometheus.auth.username }}
{{- $pass := required "Prometheus.auth.password required" .Values.prometheus.auth.password }}
---
apiVersion: v1
kind: Secret
metadata:
  name: {{ .Values.prometheus.auth.secretName }}
  namespace: {{ .Release.Namespace }}
type: Opaque
data:
  auth: {{ htpasswd $user $pass | b64enc }}
{{- end }}`

File: examples/lokahi-values.yaml
Comment (Umairk95) [inline examples/lokahi-values.yaml:1]: In prometheus and lokahi value file  username is admin and password is notset but in ./example lokahi value password is admin.

Comment (mershad-manesh) [reply examples/lokahi-values.yaml:1]: The files under the example folder are just a sample showing how you can override the default password.

**Comment by mershad-manesh** [comment]
> values are not mentioned correctly. This should be changed to
> 
> ## `{{- if not .Values.prometheus.auth.externalSecret }}
> {{- $user := required "Prometheus.auth.username required" .Values.prometheus.auth.username }}
> {{- $pass := required "Prometheus.auth.password required" .Values.prometheus.auth.password }}
> apiVersion: v1 kind: Secret metadata: name: {{ .Values.prometheus.auth.secretName }} namespace: {{ .Release.Namespace }} type: Opaque data: auth: {{ htpasswd $user $pass | b64enc }} {{- end }}`

Thanks for catching this; I have fixed it

File: dependencies/prometheus/values.yaml [hunk 1]
```diff
@@ -27,3 +27,8 @@ prometheus:
     requests:
       cpu: 100m
       memory: 100Mi
+  auth:
+    username: "admin"
+    password: "notset"
+    secretName: "prometheus-credentials"
+    externalSecret: ""
```
Comment (Umairk95) [inline dependencies/prometheus/values.yaml:34]: externalSecret should be true or false.
it is used in templates/prometheus-credentials.yaml as {{- if not .Values.auth.externalSecret  }}

File: lokahi/templates/ingress/ingresses/prometheus-ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    {{- range $Annotation, $Value := .Values.OpenNMS.ui.ingressAnnotations }}
+    {{ $Annotation | quote }}: {{ $Value | quote }}
+    {{- end }}
+    ingress.kubernetes.io/affinity: "cookie"
+    ingress.kubernetes.io/session-cookie-name: "route"
+    ingress.kubernetes.io/session-cookie-hash: "sha1"
+    nginx.ingress.kubernetes.io/ssl-redirect: "true"
+    nginx.ingress.kubernetes.io/auth-realm: Prometheus
+    nginx.ingress.kubernetes.io/auth-type: basic
+    nginx.ingress.kubernetes.io/auth-secret: {{ default "prometheus-credentials" .Values.Prometheus.auth.externalSecret }}
```
Comment (Umairk95) [inline lokahi/templates/ingress/ingresses/prometheus-ingress.yaml:17]: This should be changed to {{ default "prometheus-credentials" .Values.Prometheus.prometheus.auth.externalSecret }}

File: lokahi/templates/ingress/ingresses/prometheus-ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+spec:
+  ingressClassName: nginx
+  rules:
+    - host: {{ .Values.hostname }}
+      http:
+        paths:
+          {{- if .Values.Prometheus.enabled }}
```
Comment (Umairk95) [inline lokahi/templates/ingress/ingresses/prometheus-ingress.yaml:24]: These values should also be change.
ref:
{{- if .Values.Prometheus.enabled }} to {{- if .Values.Prometheus.prometheus.enabled }}

File: lokahi/templates/ingress/ingresses/prometheus-ingress.yaml (hunk 2 above)
Comment (mershad-manesh) [reply lokahi/templates/ingress/ingresses/prometheus-ingress.yaml:17]: If we change the it to `.Values.Prometheus.prometheus.auth.externalSecret` then the values set in Prometheus will be ignored... when user hasn't override  any of the values

File: lokahi/templates/ingress/ingresses/prometheus-ingress.yaml (hunk 3 above)
Comment (mershad-manesh) [reply lokahi/templates/ingress/ingresses/prometheus-ingress.yaml:24]: Same situation as above, see https://github.com/OpenNMS-Cloud/lokahi-helm-charts/pull/4#discussion_r1533911825

**Review by Umairk95 — APPROVED** [review]

**Review by MuhammadTalhaDataQHealth — APPROVED** [review]

**Pull request merged**